### PR TITLE
feat: integrate ai chatbot

### PR DESCRIPTION
**Pull request by Dinasoa**

### CODE REVIEW

File: src/main/java/api/prog5/bookwel/service/BookRecoService.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.ai.chat.ChatClient;
+import org.springframework.ai.chat.messages.Message;
+import org.springframework.ai.chat.prompt.Prompt;
+import org.springframework.ai.chat.prompt.PromptTemplate;
+import org.springframework.ai.chat.prompt.SystemPromptTemplate;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class BookRecoService {
```
Comment (Onitsiky) [inline src/main/java/api/prog5/bookwel/service/BookRecoService.java]: Why BookRecoService but not ChatBotService for example ?

File: src/main/java/api/prog5/bookwel/service/BookRecoService.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  private final ChatClient chatClient;
+
+  public String chatbot(String userPrompt) {
+    String systemPrompt =
```
Comment (Onitsiky) [inline src/main/java/api/prog5/bookwel/service/BookRecoService.java]: Why not defining a constant ?

File: src/main/java/api/prog5/bookwel/service/BookRecoService.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        """
+                    You are an AI Literary Recommendation Engine, providing book suggestions %s.
+                    """;
+
+    SystemPromptTemplate systemPromptTemplate = new SystemPromptTemplate(systemPrompt);
+    Message systemMessage = systemPromptTemplate.createMessage();
+
+    PromptTemplate promptTemplate =
+        new PromptTemplate(
+            """
```
Comment (Onitsiky) [inline src/main/java/api/prog5/bookwel/service/BookRecoService.java]: A constant ?

**Review by Onitsiky — CHANGES_REQUESTED** [review]

File: doc/api.yml [hunk 4]
```diff
@@ -9,6 +9,25 @@ security:
   - BearerAuth: [ ]
 
 paths:
+  '/chatbot':
```
Comment (Mahefaa) [inline doc/api.yml]: chats

File: src/main/java/api/prog5/bookwel/endpoint/rest/controller/ChatBotController.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@AllArgsConstructor
+public class ChatBotController {
+  private final ChatBotService chatBotService;
+
+  @PostMapping("/chatbot")
```
Comment (Mahefaa) [inline src/main/java/api/prog5/bookwel/endpoint/rest/controller/ChatBotController.java]: /chats

File: src/main/java/api/prog5/bookwel/endpoint/rest/controller/ChatBotController.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  public String chatBot(@RequestBody String userPrompt) {
```
Comment (Mahefaa) [inline src/main/java/api/prog5/bookwel/endpoint/rest/controller/ChatBotController.java]: chat

File: src/main/java/api/prog5/bookwel/endpoint/rest/controller/ChatBotController.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    return chatBotService.chatbot(userPrompt);
```
Comment (Mahefaa) [inline src/main/java/api/prog5/bookwel/endpoint/rest/controller/ChatBotController.java]: .chat

File: src/main/java/api/prog5/bookwel/service/ChatBotService.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.ai.chat.ChatClient;
+import org.springframework.ai.chat.messages.Message;
+import org.springframework.ai.chat.prompt.Prompt;
+import org.springframework.ai.chat.prompt.PromptTemplate;
+import org.springframework.ai.chat.prompt.SystemPromptTemplate;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class ChatBotService {
```
Comment (Mahefaa) [inline src/main/java/api/prog5/bookwel/service/ChatBotService.java]: ChattingService

File: src/main/java/api/prog5/bookwel/service/ChatBotService.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  private final ChatClient chatClient;
+
+  public String chatbot(String userPrompt) {
```
Comment (Mahefaa) [inline src/main/java/api/prog5/bookwel/service/ChatBotService.java]: chat

**Review by Mahefaa — CHANGES_REQUESTED** [review]

**Pull request merged**